src/resources/views/crud/fields/checklist.blade.php

```diff
@@ -64,7 +64,7 @@ function bpFieldInitChecklist(element) {
 
                 // set the default checked/unchecked states on checklist options
                 checkboxes.each(function(key, option) {
-                  var id = parseInt($(this).val());
+                  var id = $(this).val();
 
                   if (selected_options.includes(id)) {
```

Comment:
```suggestion
                  if (selected_options.map(String).includes(id)) {
```

When using integers for ID, this validation breaks, because `selected_options` is an array of integers, and input value is always a String.

The solution I came up with is to map that array to Strings, it's a bit heavy, but easy and readable. @pxpm do you have any better suggestion to solve this?

Reply:
The only solution I can think of is converting the numbers to strings to compare like you did. If I am not wrong `map(String)` would call `.toString`. From my knowledge (limited 🥼 ) using `number + ' '` is faster than toString. But dunno if it's worth the effort.

```javascript

//pseudo not tested

selected_options.includes(item =>function() { 
    return item + '' === id;
});
```

Reply:
In terms of performance, yes, `item + '' === id` is faster, but for the few amount of times this is used, I think we can go for the readable version `map(String)`. My question was more in the way of, should we be casting those integers to string? But yeah, I think this is the right approach 😅